core/jazz_es-kinesis-log-streamer/components/utils.js

```diff
@@ -136,96 +203,232 @@ function getApiLogsData(payload) {
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": authToken
+      },
+      method: "GET",
+      rejectUnauthorized: false
+    };
+
+    request(service_api_options, (error, response, body) => {
+      if (error) {
+        reject(error);
+      } else {
+        if (response.statusCode && response.statusCode === 200) {
+          var responseBody = JSON.parse(body);
+          logger.debug("Response Body of Service Metadata is: " + JSON.stringify(responseBody));
+          resolve(responseBody.data.services[0])
         } else {
-          data.message = message;
+          logger.debug("Service not found for this service, domain, environment: ", JSON.stringify(service_api_options));
+          resolve([])
         }
+      }
+    })
+  });
+}
 
-        data.log_level = getSubInfo(logEvent.message, config.PATTERNS.log_level, 0);
-        if (!data.log_level) {
-          data.log_level = config.DEFAULT_LOG_LEVEL;
-        }
+// Function to check if account is primary or not
+function checkIsPrimary(accountId, jsonConfig) {
+  var data = jsonConfig.config.AWS.ACCOUNTS;
+  var index = data.findIndex(x => x.ACCOUNTID == accountId);
+  if (data[index].PRIMARY) {
+    return data[index].PRIMARY;
+  } else {
+    return false;
+  }
+}
 
-        if (!(data.message.startsWith("REPORT") || data.message.startsWith("START") || data.message.startsWith("END"))) {
-          let timestmp = getSubInfo(data.message, config.PATTERNS.timestamp_pattern, 0);
-          data.message = data.message.replace(timestmp, "");
+// Function to get roleArn for specific accountId
+function getRolePlatformService(accountId, jsonConfig) {
+  var data = jsonConfig.config.AWS.ACCOUNTS;
+  var index = data.findIndex(x => x.ACCOUNTID == accountId);
+  return data[index].IAM.PLATFORMSERVICES_ROLEID;
+}
 
-          let guid = getSubInfo(data.message, config.PATTERNS.guid_regex, 0);
-          data.message = data.message.replace(guid, "");
+// Function to get configDB data
+function getConfigJson(config, token) {
+  return new Promise((resolve, reject) => {
+    var config_json_api_options = {
+      url: `${config.SERVICE_API_URL}${config.CONFIG_URL}`,
+      headers: {
+        "Content-Type": "application/json",
+        "Authorization": token
+      },
+      method: "GET",
+      rejectUnauthorized: false
+    };
+
+    request(config_json_api_options, (error, response, body) => {
+      if (error) {
+        reject(error)
+      } else {
+        if (response.statusCode && response.statusCode === 200) {
+          var responseBody = JSON.parse(body);
+          logger.debug("Response body of Config Json is: " +  JSON.stringify(responseBody));
+          resolve(responseBody.data)
+        } else {
+          logger.debug("Service not found for this service, domain, environment: ", JSON.stringify(config_json_api_options));
```

Comment:
Incorrect message. We are not getting service information here.

Reply:
Done.